base/compiler/plugin.jl

```diff
@@ -0,0 +1,47 @@
+# this simple code assumes a plugin AbstractInterpreter manages its own code cache
+# in a way that it is totally separated from the native code cache
+
+# TODO more composable way
+global PLUGIN_INTERPRETER::AbstractInterpreter = NativeInterpreter()
+isplugin(interp::AbstractInterpreter) = let
```

Comment:
Couldn't we just write
```julia
isplugin(interp::AbstractInterpreter) = false
```
and then when someone writes a plugin, they add an overload
```julia
isplugin(interp::MyInterpreter) = true
```

I tested this locally and it appears to work.

Reply:
Ah I see the problem, this doesn't work with your way of dealing with dynamic dispatch. Is that the background for https://github.com/JuliaLang/julia/issues/46220 then?